Shaded annotation processors won't work with Maven 4's `processor` type

I was curious to see if the new Maven 4 `type` items (`processor`, `modular-processor`) would work with the shaded annotation processors, and the answer seems to be no. My sample reproducer is here: https://github.com/cbarlin/shaded-processors-test

When running with both `avaje-http-jex-generator` and `avaje-http-client-generator` set to `<type>modular-processor</type>` or `<type>processor</type>`, I get:

```
[ERROR] An exception has occurred in the compiler (25). Please file a bug against the Java compiler via the Java bug reporting page (https://bugreport.java.com) after checking the Bug Database (https://bugs.java.com) for duplicates. Include your program, the following diagnostic, and the parameters passed to the Java compiler in your report. Thank you.
java.lang.LayerInstantiationException: Package io.swagger.v3.oas.models in more than one module
```

(The package it complains about changes each time it's run)

Setting them to `classpath-processor` works (although the project doesn't have Avaje Inject nor Jakarta Inject and the controller still generates an `@Singleton` which is not ideal) but that just forces Maven to use the classpath processors, rather than auto-detect (or be forced) to be used as modules.

--- a/http-generator-client/pom.xml
+++ b/http-generator-client/pom.xml
@@ -54,6 +54,22 @@
           <useModulePath>false</useModulePath>
         </configuration>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-shade-plugin</artifactId>
+        <configuration>
+          <relocations>
+            <relocation>
+              <pattern>io.avaje.http.generator.core</pattern>
+              <shadedPattern>io.avaje.http.client.generator.core</shadedPattern>
+            </relocation>
+            <relocation>
+              <pattern>io.swagger.v3</pattern>
+              <shadedPattern>io.avaje.http.client.generator.swagger.v3</shadedPattern>
+            </relocation>
+          </relocations>
+        </configuration>
+      </plugin>
     </plugins>
   </build>
 

